frontend/src/index.tsx

```diff
@@ -2,16 +2,39 @@ import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import {App} from './components/app'
 import {Home} from './components/pages/home'
-import {Students} from './components/pages/students'
+
 
 import { Router, Route, IndexRoute, Link, IndexLink, browserHistory, hashHistory  } from 'react-router'
 
+
+// If it needs async loading it will just call require.ensure (webpack)
+const component = (component : any) : any => {
+  // TODO:Removed ... and {} in return check why this is needed
+  // https://medium.com/@Nadav.Dav/implementing-lazy-loading-with-webpack-and-react-router-f8497f895892#.mq4ug0c15
+  const isReactComponent = (obj) => Boolean(obj && obj.prototype && Boolean(obj.prototype.isReactComponent));
```

Comment:
You are right, this code can be improved. This function could be use just to use always the same approach (do not differentiate whether is a component or async request), but not sure if it's worth to go that path.